src/pandas/array-test.cc

```diff
@@ -45,6 +45,177 @@ TEST_F(TestArray, Attrs) {
+      ASSERT_EQ(left_array_.data()[ii], operation(left_data_[ii], right_data_[ii]));
+    }
+    for (auto ii = 0; ii < left_array_.length(); ++ii) {
+      ASSERT_EQ(left_array_.data()[ii], result.data()[ii]);
+    }
+  }
+
+ private:
+  template <typename C_TYPE>
+  static C_TYPE* Initialize(C_TYPE (&value)[LENGTH]) {
+    for (auto ii = 0; ii < LENGTH; ++ii) {
+      // Start at 1 so that we don't get FPE with operator/
+      value[ii] = static_cast<C_TYPE>(ii + 1);
+    }
+    return value;
+  }
+
+  using LEFT_C_TYPE = typename LEFT_DATA_TYPE::c_type;
+
+  using RIGHT_C_TYPE = typename RIGHT_DATA_TYPE::c_type;
+
+  LEFT_C_TYPE left_data_[LENGTH];
+
+  RIGHT_C_TYPE right_data_[LENGTH];
+
+  std::shared_ptr<Buffer> left_buffer_;
+
+  std::shared_ptr<Buffer> right_buffer_;
+
+  LEFT_ARRAY_TYPE<LEFT_DATA_TYPE> left_array_;
+
+  RIGHT_ARRAY_TYPE<RIGHT_DATA_TYPE> right_array_;
+};
+
+TEST(TestArrayOperators, Addition) {
+  auto plus = [](auto const& left, auto const& right) { return left + right; };
+  auto plus_inplace = [](auto& left, auto const& right) { left += right; };
+
+  OperatorTest<IntegerArray, UInt8Type, IntegerArray, UInt16Type>().TestInplaceOperator(
+      plus, plus_inplace);
+  OperatorTest<IntegerArray, UInt16Type, IntegerArray, UInt8Type>().TestInplaceOperator(
+      plus, plus_inplace);
+  OperatorTest<IntegerArray, Int8Type, IntegerArray, Int16Type>().TestInplaceOperator(
+      plus, plus_inplace);
+  OperatorTest<IntegerArray, Int16Type, IntegerArray, Int8Type>().TestInplaceOperator(
+      plus, plus_inplace);
+  OperatorTest<IntegerArray, UInt32Type, IntegerArray, UInt16Type>().TestInplaceOperator(
+      plus, plus_inplace);
+  OperatorTest<IntegerArray, UInt64Type, IntegerArray, UInt32Type>().TestInplaceOperator(
+      plus, plus_inplace);
+  OperatorTest<IntegerArray, Int32Type, IntegerArray, Int64Type>().TestInplaceOperator(
+      plus, plus_inplace);
+  OperatorTest<IntegerArray, Int64Type, IntegerArray, Int32Type>().TestInplaceOperator(
+      plus, plus_inplace);
+
+  OperatorTest<IntegerArray, Int8Type, IntegerArray, Int8Type>().TestInplaceOperator(
+      plus, plus_inplace);
+  OperatorTest<IntegerArray, Int16Type, IntegerArray, Int16Type>().TestInplaceOperator(
+      plus, plus_inplace);
+  OperatorTest<IntegerArray, Int32Type, IntegerArray, Int32Type>().TestInplaceOperator(
+      plus, plus_inplace);
+  OperatorTest<IntegerArray, Int64Type, IntegerArray, Int64Type>().TestInplaceOperator(
+      plus, plus_inplace);
+  OperatorTest<IntegerArray, UInt8Type, IntegerArray, UInt8Type>().TestInplaceOperator(
+      plus, plus_inplace);
+  OperatorTest<IntegerArray, UInt16Type, IntegerArray, UInt16Type>().TestInplaceOperator(
+      plus, plus_inplace);
+  OperatorTest<IntegerArray, UInt32Type, IntegerArray, UInt32Type>().TestInplaceOperator(
+      plus, plus_inplace);
+  OperatorTest<IntegerArray, UInt64Type, IntegerArray, UInt64Type>().TestInplaceOperator(
```

Comment:
Guess we may want some macros (or other templates) to be able to say "Do this thing for the cartesian product of certain types"